WordPress/Classes/Services/PostService.m

```diff
@@ -333,70 +337,129 @@ - (void)autoSave:(AbstractPost *)post
             failure(error);
         }];
     };
+}
 
-    id<PostServiceRemote> remote = [self.postServiceRemoteFactory forBlog:post.blog];
-    if ([remote isKindOfClass:[PostServiceRemoteREST class]]) {
-        PostServiceRemoteREST *restRemote = (PostServiceRemoteREST*) remote;
-        RemotePost *remotePost = [self remotePostWithPost:post];
-
-        void (^successBlock)(RemotePost *post, NSString *previewURL) = ^(RemotePost *post, NSString *previewURL) {
-            [self.managedObjectContext performBlock:^{
-                AbstractPost *postInContext = (AbstractPost *)[self.managedObjectContext existingObjectWithID:postObjectID error:nil];
-                if (postInContext) {
-                    postInContext.remoteStatus = AbstractPostRemoteStatusAutoSaved;
-                    [self updateMediaForPost:postInContext success:^{
-                        if (success) {
-                            success(postInContext, previewURL);
-                        }
-                    } failure:^(NSError * _Nullable error) {
-                        DDLogInfo(@"Error in updateMediaForPost while remote auto-saving post. description: %@", error.localizedDescription);
-                        // even if media fails to attach we are answering with success because the post auto-save was successful.
-                        if (success) {
-                            success(postInContext, previewURL);
-                        }
-                    }];
-                } else {
-                    // This can happen if the post was deleted right after triggering the auto-save.
+- (AutosaveSuccessBlock)wrappedAutosaveSuccessBlock:(NSManagedObjectID *)postObjectID
+                                            success:(void (^)(AbstractPost *post, NSString *previewURL))success
+{
+    return ^(RemotePost *post, NSString *previewURL) {
```

Comment:
It looks like we never actually used `post` in the old code. 😅 Probably not worth it to change right now though.